### PR TITLE
Change UnMock task to Gradle's dependency transform

### LINKED ISSUE #82: processing android jar once on multi-module project
Hello,

we have noticed that on the multi-module project, where multiple modules contain unmock plugin (with identical configuration), android jar needs to be processed for each module separately.

Is there any way to do it only once and reuse it on different modules?

### PR DESCRIPTION
**Pull request by matejdro**
It seems like Unmock is doing exactly what Gradle's [dependency transformation system](https://docs.gradle.org/current/userguide/artifact_transforms.html) was designed to do. So this PR moves Unmock from being a task to a dependency transformer, which allows across-module caching.

This fixes #82.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-all.zip
```
Comment (bojankseneman) [inline gradle/wrapper/gradle-wrapper.properties:3]: we could switch to binary artifact, I don't see any reason why all is needed

**Review by bjoernQ — APPROVED** [review]
LGTM - Thanks! Going to merge this and hopefully I can publish a new version next weekend

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (matejdro) [reply gradle/wrapper/gradle-wrapper.properties:3]: It was all before, so I did not want to change it. Plus, from what I understand, all includes sources/javadocs, which makes IDE experience a bit better when interacting with Gradle scripts.

**Comment by matejdro** [comment]
Thanks!

**Pull request merged**